cpp/oneapi/dal/algo/connected_components/backend/cpu/vertex_partitioning_default_kernel.hpp

```diff
@@ -20,16 +20,157 @@
 #include "oneapi/dal/backend/common.hpp"
 #include "oneapi/dal/backend/memory.hpp"
 #include "oneapi/dal/backend/interop/common.hpp"
-#include "oneapi/dal/detail/error_messages.hpp"
+#include "oneapi/dal/table/homogen.hpp"
+#include "oneapi/dal/backend/primitives/rng/rng_engine.hpp"
 
 namespace oneapi::dal::preview::connected_components::backend {
+using namespace oneapi::dal::preview::detail;
+using namespace oneapi::dal::preview::backend;
+
+inline void link(const std::int32_t &u, const std::int32_t &v, std::int32_t *components) {
+    std::int32_t p1 = components[u];
+    std::int32_t p2 = components[v];
+    std::int32_t h;
+    std::int32_t l;
+    while (p1 != p2) {
+        if (p1 > p2) {
+            h = p1;
+            l = p2;
+        }
+        else {
+            h = p2;
+            l = p1;
+        }
+        if (components[h] == h) {
+            components[h] = l;
+            break;
+        }
+        p1 = components[components[h]];
+        p2 = components[l];
+    }
+}
+
+inline void compress(const std::int32_t &u, std::int32_t *components) {
+    while (components[components[u]] != components[u]) {
+        components[u] = components[components[u]];
+    }
+}
+
+inline void order_component_ids(const std::int64_t &vertex_count,
```

Comment:
Have you analyzed if this function was actually inlined? If not analyzed, or  it was not inlined -> add template CPU.